packages/transport/src/chrome-runtime/background-connection-manager.ts

```diff
@@ -72,105 +77,163 @@ export class BackgroundConnectionManager {
         port: clientPort,
         sender,
         streams: new Map<string, AbortController>(),
-      } as BackgroundConnection;
+        tlsChannelId,
+        documentId,
+      };
 
       /**
-       * This method is used to generate a handler for a connection request from
-       * the client to open a sub-channel, as instructed by this connection
-       * manager. The sub-channel encapsulates a response stream.
+       * This method is used to manage a sub-channel, encapsulating a response
+       * stream. A TransportInitChannel representing the channel is returned for
+       * transport to the client. The client should respond by opening a
+       * connection bearing this name.
        *
-       * @param subStream ReadableStream to be transmitted in a sub-channel
-       * @param subName Name of a chrome.runtime.port representing a sub-channel
+       * @param subStream ReadableStream to be transmitted via sub-channel
        */
-      const subHandler = (subStream: ReadableStream<JsonValue>, subName: string) => {
+      const initSubChannel = ({ requestId, stream }: TransportStream): TransportInitChannel => {
+        const [channel] = nameChannel(ChannelSubLabel.ServerStream);
+        const acont = new AbortController();
+
         /**
-         * This is the generated connection listener. It checks for a specified
-         * name, which is transmitted to the client by the caller. Any
-         * connecting port not identifying that name is ignored.
+         * This is the generated connection listener. It checks for our
+         * generated subName, and any connection not bearing the name is
+         * ignored.  The sink transmits the stream piecewise in the channel.
          *
          * @param subPort chrome.runtime.Port, initiated by any new runtime connection
          */
-        const subConnectionListener = (subPort: chrome.runtime.Port) => {
-          if (subPort.name !== subName) return;
-          chrome.runtime.onConnect.removeListener(subConnectionListener);
-          const acont = new AbortController();
-          connection.streams.set(subName, acont);
-          subPort.onDisconnect.addListener(() => acont.abort(subName));
-          void subStream
+        const subListener = (subPort: chrome.runtime.Port) => {
+          if (subPort.name !== channel) return;
+          chrome.runtime.onConnect.removeListener(subListener);
+          connection.streams.set(channel, acont);
+          subPort.onDisconnect.addListener(() => acont.abort(channel));
+          void stream
             .pipeTo(new WritableStream(new ChromeRuntimeStreamSink(subPort)), {
               signal: acont.signal,
             })
-            .catch(e => {
-              if (e !== subName) throw e;
+            .catch((error: unknown) => {
+              if (error !== channel) {
+                // TODO: handling errors in this promise chain is problematic.
+                // we're in a callback, and can't reliably answer the requester.
```

Comment:
But isn't the postMessage below answering the requestor? When you say it's not reliable, does that mean it only sends sometimes?